packages/cactus-plugin-keychain-vault/src/main/typescript/plugin-keychain-vault.ts

```diff
@@ -272,14 +305,134 @@ export class PluginKeychainVault implements IPluginWebService, IPluginKeychain {
+
+  /**
+   * @description send message digest to be signed by private key stored on vault
+   * @param digest : messages digest which need to signed
+   * @param preHashed : is digest already hashed
+   * @returns asn1 encoded signature
+   */
+  async transitSign(
+    keyName: string,
+    digest: Buffer,
+    preHashed: boolean,
+  ): Promise<Buffer> {
+    if (!this.transitBackend) {
+      throw new Error("transit vault secret engine not supported");
+    }
+    const fnTag = `${this.className}#transitSign`;
+    this.log.debug(
+      `${fnTag} sign message digest with size = ${digest.length} , preHashed = ${preHashed} with key = ${keyName}`,
+    );
+    try {
+      const resp = await this.transitBackend.write("sign/" + keyName, {
+        input: digest.toString("base64"),
+        prehashed: preHashed,
+        marshaling_algorithm: "asn1",
+      });
+      this.log.debug(`Response from Vault: %o`, JSON.stringify(resp));
+      if (resp?.data?.signature) {
+        const base64Sig = (resp.data.signature as string).split(":")[2];
+        return Buffer.from(base64Sig, "base64");
+      } else {
+        throw new Error(
+          `${fnTag}: Invalid response received from Vault. Expected "resp.data.signature" property chain to be truthy`,
+        );
+      }
+    } catch (error) {
+      if ((error as any)?.response?.statusCode === HttpStatus.NOT_FOUND) {
+        throw new Error("signing key not found");
+      }
+      throw error;
+    }
+  }
+
+  /**
+   * @description will generate a a private key, new if key doesn't exists , and does nothing if key already exists
+   * @param keyName label of the key
+   * @param keyType
+   */
+  async transitNewKey(
+    keyName: string,
+    keyType: VaultTransitKeyType,
+  ): Promise<void> {
+    if (!this.transitBackend) {
+      throw new Error("transit vault secret engine not supported");
+    }
+    const fnTag = `${this.className}#transitNewKey`;
+    this.log.debug(`${fnTag} keyName = ${keyName} of type ${keyType}`);
+    await this.transitBackend.write("keys/" + keyName, {
+      type: keyType,
+    });
+  }
+
+  /**
+   * @description return public key of latest version
+   * @param keyName for which public key should be returned
+   * @returns jwk encoded public key
+   */
+  async transitGetPub(keyName: string): Promise<KeyObject> {
+    if (!this.transitBackend) {
+      throw new Error("transit vault secret engine not supported");
+    }
+    const fnTag = `${this.className}#transitGetPub`;
+    this.log.debug(`${fnTag} keyName = ${keyName}`);
+    try {
+      const resp = await this.transitBackend.read("keys/" + keyName);
+      this.log.debug(`Response from Vault: %o`, JSON.stringify(resp));
+      if (resp?.data?.latest_version && resp?.data?.keys) {
+        if (
+          !Object.values(VaultTransitKeyType).includes(
+            resp.data.type as VaultTransitKeyType,
+          )
+        ) {
+          throw new Error("invalid public key type");
```

Comment:
@Zzocker Please also include the `fnTag` in the error message. Rationale: If someone calls the API and this message will be the only thing they see in an error response, we should provide them with a little more context on where/what went wrong. In the same line of thinking I also highly recommend to include the `keyName` in the error message.